src/main/java/com/bugsnag/android/Client.java

```diff
@@ -166,14 +171,29 @@ public Client(@NonNull Context androidContext, @NonNull Configuration configurat
         }
 
         // register a receiver for automatic breadcrumbs
-        androidContext.registerReceiver(eventReceiver, EventReceiver.getIntentFilter());
+        appContext.registerReceiver(eventReceiver, EventReceiver.getIntentFilter());
         config.addObserver(this);
 
         // Flush any on-disk errors
         errorStore.flush(errorReportApiClient);
 
         boolean isNotProduction = !AppData.RELEASE_STAGE_PRODUCTION.equals(AppData.guessReleaseStage(appContext));
         Logger.setEnabled(isNotProduction);
+
+        appContext.registerReceiver(new ConnectivityChangeReceiver(), new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION));
+    }
+
+    private class ConnectivityChangeReceiver extends BroadcastReceiver {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+            NetworkInfo networkInfo = cm.getActiveNetworkInfo();
+            boolean retryReports = networkInfo != null && networkInfo.isConnectedOrConnecting();
+
+            if (retryReports) {
+                errorStore.flush(errorReportApiClient);
```

Comment:
Is there any risk here for duplicate reports if the error store is being flushed when this is called?